subprojects/core-api/src/main/java/org/gradle/api/tasks/util/internal/PatternSpecFactory.java

```diff
@@ -99,7 +100,7 @@ private synchronized Spec<FileTreeElement> getDefaultExcludeSpec(CaseSensitivity
     }
 
     private boolean invalidChangeOfExcludes(String[] defaultExcludes) {
-        return !Arrays.equals(previousDefaultExcludes, defaultExcludes);
+        return !(new HashSet(Arrays.asList(previousDefaultExcludes))).equals(new HashSet(Arrays.asList(defaultExcludes)));
```

Comment:
❌ Instead of converting `defaultExcludes` and `previousDefaultExcludes` to `HashSet` here, please change their types to `HashSet` since these values are reused.  
See `previousDefaultExcludes` [on the line 50](https://github.com/gradle/gradle/pull/26452/files#diff-6f5daae9e2ab6c4bfa6d93e1023146d63e8810ace5b4b8c247a38f53595735f9R50) and `defaultExcludes`. This would also affect `updateDefaultExcludeSpecCache`, `setDefaultExcludesFromSettings`, and `failOnChangedDefaultExcludes`.

Reply:
Thanks for your reply! I addressed the change request.